### PR TITLE
GH-937 Setting Table Styles

### PR DESCRIPTION
**Pull request by dtsnell4**
Changes the creation of the fixed Column from native Javascript to D3.js.  Also fixes highlighting of table rows when fixedColumns() is true and showHeader() is false.

Fixes GH-937, GH-936

Signed-off-by: Dan Snell <Dan.Snell@lexisnexis.com>

### CODE REVIEW

File: src/other/Table.js [hunk 1]
```diff
@@ -424,45 +424,72 @@
...
-                    var tempDiv = document.createElement("div");
-                    tempDiv.appendChild(td);
-                    rowContents += "<tr>" + tempDiv.innerHTML + "</tr>";
-                });
-                rowLabelsWrapper.html(rowContents)
+                    tHead.enter()
+                        .insert("thead", "tbody")
+                        .append("tr")
+                        .style("background-color", context.theadRowBackgroundColor())
+                    ;
```
Comment (GordonSmith) [inline src/other/Table.js]: You could have made this:
```
var tHead_tr = tHead.enter()
                        .insert("thead", "tbody")
                        .append("tr")
                        .style("background-color", context.theadRowBackgroundColor())
                    ;
```
Which would then simplify the selectAll below (just a FYI).

File: src/other/Table.js [hunk 2]
```diff
@@ -424,45 +424,72 @@
...
+                    tHead.exit()
+                        .remove()
+                    ;
+
+                    var ths =  tHead.selectAll("tr").selectAll("th").data(origThead.filter(function (d, idx) { return idx === 0; }));
```
Comment (GordonSmith) [inline src/other/Table.js]: So this could then be:
```
var ths =  tHead_tr.selectAll("th").data(...
```

**Comment by GordonSmith** [comment]
@dtsnell4 - Actually this isn't going to work as it stands - when I edit the "columns" property, it just gets messed up...

File: src/other/Table.js [hunk 3]
```diff
@@ -424,45 +424,72 @@
...
-                }
-                rowContents += "</thead>";
-                rowSelection.each(function () {
-                    var td = document.createElement("td");
-                    td.className = "row-label";
-                    td.style.color = context.tbodyFontColor();
-                    td.style.fontSize = context.tbodyFontSize();
-                    td.style.fontFamily = context.tbodyFontFamily();
-                    td.style.borderColor = context.tbodyCellBorderColor();
-                    td.innerHTML = this.innerHTML ? this.innerHTML : "&nbsp;";
```
Comment (GordonSmith) [inline src/other/Table.js]: The &nbsp; is import as it stops the rows going to 0 height.

**Comment by dtsnell4** [comment]
@GordonSmith ready for re-review

**Pull request merged**